src/wallet/rpcwallet.cpp

```diff
@@ -2684,14 +2684,15 @@ static UniValue createwallet(const JSONRPCRequest& request)
 
     std::string error;
     std::string warning;
-    WalletCreationStatus status;
-    std::shared_ptr<CWallet> wallet = CreateWallet(*g_rpc_interfaces->chain, request.params[0].get_str(), error, warning, status, passphrase, flags);
-    if (status == WalletCreationStatus::CREATION_FAILED) {
-        throw JSONRPCError(RPC_WALLET_ERROR, error);
-    } else if (status == WalletCreationStatus::ENCRYPTION_FAILED) {
-        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, error);
-    } else if (status != WalletCreationStatus::SUCCESS) {
-        throw JSONRPCError(RPC_WALLET_ERROR, "Wallet creation failed");
```

Comment:
Should keep this by adding a `default:`?

Reply:
Sjors suggestion was to go without a `default` so that the compiler will show a warning if we forget to handle any newly added options in the WalletCreationStatus enum. The same technique is used in the same file here: https://github.com/fjahr/bitcoin/blob/07df41c56bc3c0ccb4463083febf39fcad9749e2/src/wallet/rpcwallet.cpp#L3485

But I will add a comment, same as in the example mentioned above.

Reply:
A warning can be unnoticed.

Reply:
Then, it should be made an error

Reply:
```
$ ./configure --help|grep enable-werror
  --enable-werror         Treat certain compiler warnings as errors (default
$ git grep enable-werror .travis.yml
.travis.yml:        BITCOIN_CONFIG="--enable-gui --enable-reduce-exports --enable-werror"

Reply:
@promag Here you go:

* build: Treat -Wswitch as error when --enable-werror #16424